scripts/build.py

```diff
@@ -380,27 +488,34 @@ def hash_should_be_inserted(filepath):
                    in FILEPATHS_NOT_TO_RENAME)
 
 
-def copy_files_source_to_target(source, target, file_hashes):
-    """Copies all files in source directory to target directory, renames
-    them to include hash in their name.
+def generate_copy_tasks_to_copy_from_source_to_target(
+        source, target, file_hashes):
+    """Generate copy task for each file in source directory, excluding files
+    with extensions in FILE_EXTENSIONS_TO_IGNORE. Insert hash from hash dict
+    into the destination filename.
 
     Args:
         source: str. Path relative to /oppia directory of directory
             containing files and directories to be copied.
         target: str. Path relative to /oppia directory of directory where
             to copy the files and directories.
-        file_hashes: dict(str, str). Dictionary of file hashes.
+        file_hashes: dict(str, str). Dictionary with filepaths as keys and
+            hashes of file content as values.
+
+    Returns:
+        copy_tasks: deque(Thread). A deque that contains all copy tasks queued
+            to be processed.
     """
     print 'Processing %s' % os.path.join(os.getcwd(), source)
     print 'Copying into %s' % os.path.join(os.getcwd(), target)
-
+    copy_tasks = collections.deque()
     ensure_directory_exists(target)
     shutil.rmtree(target)
```

Comment:
Oooh, this has side-effects. Doesn't seem to be reflected in the function name though. Maybe we should move it outside this function?

In general, might want to move shutil.rmtree into its own function so we can stub it out in tests.

Reply:
Move `ensure_directory_exists` and `rmtree` into `safe_delete_directory_tree` function